### PR TITLE
Updating CONTRIBUTING.md to explain when to update schema version

### PR DESCRIPTION
**Pull request by nickkaczmarek**
## Checklist:
- [x] Did you sign the [Contributor License Agreement](https://cla-assistant.io/wwt/WorkflowSchema)?
- [x] Did you add the appropriate labels to the PR?

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -4,6 +4,26 @@ Thank you for your interest in WorkflowSchema!
 [![Issues](https://img.shields.io/github/issues/wwt/WorkflowSchema?color=bright-green)](https://github.com/wwt/WorkflowSchema/issues)
 [![Pull Requests](https://img.shields.io/github/issues-pr/wwt/WorkflowSchema?color=bright-green)](https://github.com/wwt/WorkflowSchema/pulls)
 
+## Updating the schema
+If your contribution meets any of the below guidelines it's required that you add the appropriate label on the pull request.
+Using these labels will run a GitHub action that revs the version accordingly.
```
Comment (morganzellers) [inline .github/CONTRIBUTING.md:9]:
```suggestion
Using these labels will run a GitHub action that updates the version accordingly.
```

**Review by morganzellers — CHANGES_REQUESTED** [review]

**Review by morganzellers — DISMISSED** [review]

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -4,6 +4,26 @@ Thank you for your interest in WorkflowSchema!
...
+
+#### Important note
+Updating the version of the schema requires that all the libraries using the schema are updated to reflect the new version.
+This includes [SwiftCurrent](https://github.com/wwt/SwiftCurrent).
+
+### No label needed
```
Comment (Tyler-Keith-Thompson) [inline .github/CONTRIBUTING.md:15]: Suggest we add examples of where it would and would not change.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -4,6 +4,26 @@ Thank you for your interest in WorkflowSchema!
...
+- metadata changes. `ex. examples are updated, descriptions are revamped, etc.`
```
Comment (Tyler-Keith-Thompson) [inline .github/CONTRIBUTING.md:16]: DO NOT VERSION: When there are just metadata changes that would not affect any parser in any library
DO NOT VERISON: When examples are updated
DO NOT VERSION: When descriptions are revamped

DO VERSION: When any field that is parsed changed
DO VERSION: When new fields are added
DO VERSION: When the validation requirements for a field change (minLength > 0, etc...)

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -4,6 +4,26 @@ Thank you for your interest in WorkflowSchema!
...
+
+### Patch label
+- additive changes `ex. new fields`
```
Comment (Tyler-Keith-Thompson) [inline .github/CONTRIBUTING.md:19]: DO PATCH: When new fields are added
DO PATCH: When validation requirements for a field change (minLength > 0)

DO NOT PATCH: When a field is parsable both as a string and as an object
DO NOT PATCH: When a field name is changed
DO NOT PATCH: When a field type is changed

**Review by morganzellers — APPROVED** [review]

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -4,6 +4,26 @@ Thank you for your interest in WorkflowSchema!
...
+
+### Minor label
+- break to the existing api. `ex. changing the way that FlowRepresentableName works`
```
Comment (Tyler-Keith-Thompson) [inline .github/CONTRIBUTING.md:22]: DO MINOR: When a field is made parsable both as a string and as an object
DO MINOR: When a field name is changed
DO MINOR: When a field type is changed

DO NOT MINOR: When we change a fundamental behavior (flowPersistence)
DO NOT MINOR: When `sequence` is no longer what we use to describe workflows

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -4,6 +4,26 @@ Thank you for your interest in WorkflowSchema!
...
+
+### Major label
+- fundamental shift in design. `ex. [sequence] goes away and we decide on a new way to describe workflows`
```
Comment (Tyler-Keith-Thompson) [inline .github/CONTRIBUTING.md:25]: DO MAJOR: When we change a fundamental behavior (flowPersistence)
DO MAJOR:  When `sequence` is no longer what we use to describe workflows

**Review by Tyler-Keith-Thompson — CHANGES_REQUESTED** [review]
I didn't add these as suggestions because I don't want them taken verbatim. However, they are heavily inspired by our styleguide and meant to clarify ambiguity. I think conceptually it's right. I think terminology needs more thought than I put into it for this review.

**Pull request merged**